src/routes/safe/components/Settings/DataExport/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { ReactElement } from 'react'
+import { useSelector } from 'react-redux'
+import Button from 'src/components/layout/Button'
+import { addressBookState } from 'src/logic/addressBook/store/selectors'
+import { SafeRecordWithNames, safesWithNamesAsList } from 'src/logic/safe/store/selectors'
+import { AddressBookState } from 'src/logic/addressBook/model/addressBook'
+import { isValidAddress } from 'src/utils/isValidAddress'
+
+// Returns address book data in the format of web-core
+const parseAddressBook = (addressBook: AddressBookState) => {
+  const newAb = addressBook.reduce((acc, { address, name, chainId }) => {
+    if (!name || !address || !isValidAddress(address)) {
+      return acc
+    }
+    acc[chainId] = acc[chainId] || {}
+    acc[chainId][address] = name
+    return acc
+  }, {})
+
+  return Object.keys(newAb).length > 0 ? newAb : {}
+}
+
+// Returns added safe data in the format of web-core
+const parseAddedSafes = (safes: SafeRecordWithNames[]) => {
+  const newAddedSafes = {}
+
+  for (const safe of safes) {
+    if (!safe.chainId) {
+      continue
+    }
+
+    if (!newAddedSafes[safe.chainId]) {
+      newAddedSafes[safe.chainId] = {}
+    }
+
+    newAddedSafes[safe.chainId][safe.address] = {
+      owners: safe.owners,
+      threshold: safe.threshold,
+      ethBalance: safe.ethBalance,
+    }
+  }
+
+  return newAddedSafes
+}
+
+const DataExport = (): ReactElement => {
+  const addressBook = useSelector(addressBookState)
+  const safes = useSelector(safesWithNamesAsList)
+
+  const handleExport = () => {
+    const filename = `safe-data-${new Date().toISOString().slice(0, 10)}.json`
+
+    const ABData = parseAddressBook(addressBook)
+    const safesData = parseAddedSafes(safes)
+
+    const data = JSON.stringify({
+      SAFE_v2__addedSafes: safesData,
+      SAFE_v2__addressBook: ABData,
```

Comment:
I think we should keep these properties more generic, e.g. the names of the relevant slices. Importing these should be handled by the reducers in `web-core`, not setting the `localStorage` directly.